nftables_test.go

```diff
@@ -74,6 +81,44 @@ func ifname(n string) []byte {
 	return b
 }
 
+// openSystemNFTConn returns a netlink connection that tests against
+// the running kernel in a separate network namespace.
+// cleanupSystemNFTConn() must be called from a defer to cleanup
+// side effects of this function.
+func openSystemNFTConn(t *testing.T) (*nftables.Conn, netns.NsHandle, netns.NsHandle) {
+	t.Helper()
+	if !*enableSysTests {
+		t.SkipNow()
+	}
+	runtime.LockOSThread()
+
+	// Get the current namespace, so we can switch
```

Comment:
Instead of the rather elaborate switching back and forth, could you please refactor the dial calls (e.g. https://github.com/google/nftables/blob/ce006eb26de8c3158d9d43bfee697c9aac5208a3/nftables.go#L572-L575) into a helper function which specifies a https://godoc.org/github.com/mdlayher/netlink#Config with NetNS set?

Reply:
Sure thing. Given that running this in a network namespace could be a useful feature anyway, ive added the `NetNS` field to allow users to specify a network namespace.

Reply:
I've left the thread locking as IIRC unshare syscall is a task-level operation. Could be wrong thou.